net/calendar-contacts-server/Portfile

```diff
@@ -0,0 +1,545 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               github 1.0
+
+github.setup            apple ccs-calendarserver fbd0e11
+git.branch              master
+
+name                    calendar-contacts-server
+# version from https://github.com/apple/ccs-calendarserver/blob/master/setup.py
+version                 9.3
+revision                0
+categories              net mail
+platforms               darwin
+supported_archs         noarch
+maintainers             {ieee.org:s.t.smith @essandess} openmaintainer
+license                 Apache
+
+description             Apple Calendar and Contacts Server configuration
+long_description ${description}. \
+    ccs-calendarserver is a standards-compliant server implementing \
+    the CalDAV and CardDAV protocols, including iMIP support. It \
+    provides a shared location on the network allowing multiple users \
+    to store and edit calendaring and contact information. This port \
+    provides a basic, working, easily modifiable configuration, \
+    previously used in macOS Server.app, and an nginx reverse proxy \
+    to handle modern crypto and isolate the backend server.
+
+homepage                https://www.calendarserver.org
+
+checksums               rmd160  821ba4d143320768abdbc349f7cd25be02126ab1 \
+                        sha256  475d444e769c2c1be270743d583a43215ae90c061bb51171ae3cbb9914207c61 \
+                        size    3713755
+
+# use these to specify python versions, python2 required
+set python2_version 2.7
+set python2_version_nickname \
+    [join [lrange [split ${python2_version} .] 0 1] {}]
+
+# use these to specify PostgreSQL versions, postgresql9 required
+set postgresql9_version 9.6
+set postgresql9_version_nickname \
+    [join [lrange [split ${postgresql9_version} .] 0 1] {}]
+
+depends_lib-append      port:cyrus-sasl2 \
+                        port:libffi \
+                        port:mail-server \
+                        port:memcached \
+                        port:nginx \
+                        port:openssl \
+                        port:postgresql${postgresql9_version_nickname}-server \
+                        port:python${python2_version_nickname} \
+                        port:py${python2_version_nickname}-pip \
+                        port:py${python2_version_nickname}-pyobjc-cocoa
+
+depends_run-append      port:pip_select \
+                        port:postgresql_select
+
+variant initialize \
+    description {Initialize all configuration files. Existing
+        configurations files are not overwritten by default.} {}
```

Comment:
Be sure to consider what happens at upgrade time. Considering the upstream package isn't really moving very fast these days (or at all), it might not be worth spending a ton of time on a polished upgrade experience. The main thing is to make it hard to accidentally delete or overwrite one's own data or configs.

Reply:
Acknowledged. This variant will never necessary for most users, unless they want to reset their config files to factory. It isn't even necessary on the first install.

The purpose of this is really just for development and a hard factory reset if necessary. The Portfile logic prevents loss of user configurations unless the user is really trying hard to blow them away.

I added language to clarify all this: in the variant description, a `ui_warn`, and in the notes.

I thought about a shell script to do this as @ryandesign suggests, but I like the convenience of just adding this variant to the `port install` command.

Please push back if you feel strongly, as I've used this approach in other ports and plan to update them per consensus in this PR.